test/application/use.js

```diff
@@ -70,60 +65,50 @@ describe('app.use(fn)', () => {
 
     const server = app.listen();
 
-    request(server)
-      .get('/')
-      .expect(404)
-      .end(err => {
-        if (err) return done(err);
-        calls.should.eql([1, 2, 3, 4, 5, 6]);
-        done();
-      });
+    const res = await request(server, '/');
+    expect(res.status).toBe(404);
```

Comment:
assert +1, don't like expect either, maybe we can try [power-assert](https://github.com/power-assert-js/power-assert) later, it has the same API with node's assert.